COMPONENT-GUIDELINES.md

```diff
@@ -0,0 +1,293 @@
+# Purpose
+
+This document defines guidelines and patterns developers should follow when building components for the Pipedream registry.
+
+Developers may create, deploy and share components that do not conform to these guidelines, but they will not be eligible to be listed in the curated registry (e.g., they may be hosted in a Github repo). If you develop a component that does not adhere to these guidelines, but you believe there is value to the broader community, please [reach out in our community forum](https://pipedream.com/community). We can either provide guidance to improve compliance or evaluate if an exception is warranted.
+
+# Overview
+
+[Pipedream](https://pipedream.com) is a low code integration platform that makes it easy to connect APIs remarkably fast. Users can select from thousands of customizable, open source components for hundreds of apps, and orchestrate their execution in workflows. Developers can [contribute](CONTRIBUTING.md) to these open source components on [Github](https://github.com/pipedreamhq/pipedream) by:
+
+- Creating new comonents (sources and actions)
+- Updating exisiting components (e.g., fixing bugs, enhancing functionality)
+- Adding or updating metadata (e.g., descriptions, labels)
+
+Once a PR is merged to the `master` branch of the `pipedreamhq/pipedream` repo, the components are automatically registered and immediately become available to the 150k+ users of the Pipedream platform. 
+
+## Components
+
+Components are [Node.js modules](COMPONENT-API.md#component-structure) that run on Pipedream's serverless infrastructure. They may use Pipedream managed auth for 300+ apps and use most npm packages with no `npm install` or `package.json` required. Pipedream currently supports two types of components — sources and actions.
+
+### Sources
+
+- Emit events that can trigger Pipedream workflows (events may also be consumed outside of Pipedream via API)
+- Emitted event data can be inspected and referenced by steps in the target workflow
+- Can use any of Pipedream's built-in deduping strategies
+- Can be triggered on HTTP requests, timers, cron schedules, or manually
+- May store and retrieve state using the [built-in key-value store](https://github.com/PipedreamHQ/pipedream/blob/master/COMPONENT-API.md#db)
+
+### Actions
+
+- May be used as steps in workflows to perform common functions (e.g., get or modify data in an app)
+- Data returned by actions may be inspected and used in future workflow steps
+
+## Pipedream Registry
+
+The Pipedream registry consists of sources and actions that have been curated for the community. Registered components are verified by Pipedream through the Github PR process, and:
```

Comment:
Grammar nitpick: There doesn't need to be a comma here: "through the Github PR process and:"